test/test_va_api_fixture.cpp

```diff
@@ -42,13 +48,18 @@ VAStatus VAAPIFixtureSharedDisplay::s_initStatus = VA_STATUS_SUCCESS;
 VAAPIFixture::VAAPIFixture()
     : ::testing::Test::Test()
     , m_vaDisplay(NULL)
-    , m_restoreDriverName(getenv("LIBVA_DRIVER_NAME"))
     , m_drmHandle(-1)
     , m_configID(VA_INVALID_ID)
     , m_contextID(VA_INVALID_ID)
     , m_bufferID(VA_INVALID_ID)
     , m_skip("")
 {
+    memset(&m_restoreDriverName[0], 0, sizeof(m_restoreDriverName));
```

Comment:
We're changing from `char *` to a fixed size string. Might be wroth a separate commit with some details why

Reply:
Final nit - if it were me, I would prefix the commits with the section they touch.

For example:
vainfo: add va-win32 support
tests: remove non C++ standard constructs
tests: enable building on Windows

Reply:
I broke down the commits more for clarity :)

Reply:
> It is not safe to modify the value of the environment variable using the returned pointer

We don't do that, do we?

Reply:
Yes, in `VAAPIFixture::~VAAPIFixture` it restores the value using `setenv("LIBVA_DRIVER_NAME", m_restoreDriverName, 1);`.
Passing `m_restoreDriverName` to `setenv` with it being the exact pointer returned from `getenv` produces sets the environment variable to garbage (I can confirm running the tests when vainfo prints the driver name in screen)

Reply:
According to `man setenv`
```
int setenv(const char *name, const char *value, int overwrite);
```

In other words - the string is not modified and the current comment is misleading at best.

Perhaps Windows `setenv()` modifies it, or it has different lifetime there? Don't think the "why" matters too much - just document that you're seeing garbage on Windows and let's merge this already - hint maintainers, hint :wink: :smiling_imp: